lib/qs/payload.rb

```diff
@@ -1,31 +1,75 @@
 require 'qs'
+require 'qs/event'
 require 'qs/job'
 
 module Qs
 
   module Payload
 
+    VALID_PAYLOAD_TYPES = [
+      Job::PAYLOAD_TYPE,
+      Event::PAYLOAD_TYPE
+    ].freeze
```

Comment:
@jcredding we discussed altering this and making it a with payload type keys that point to valid method names.  The hash would raise a friendly error for any unknown keys.  Then you can take out the custom error handling and checking validation.